newspeak/MirrorsForPrimordialSoup.ns

```diff
@@ -796,6 +740,16 @@ public preamble ^<Symbol> = (
 	headerAst:: (parsing Parser for: source) advanceToken parseClassHeader.
 	^source copyFrom: headerAst start to: headerAst superConstructorCall end
 )
+public primaryFactorySelector ^ <Symbol> = (
+  ^mixin _primaryFactorySelector
+)
+public selectors ^ <List[Symbol]> = (
+  | result <List[Symbol]> = List new. |
+  instanceInitializer literals do: [:s |
+	 s isKindOfString ifTrue: [result add: (s copyFrom: (s lastIndexOf: '`') + 1 to: s size)]
+	].
```

Comment:
Yes. In fact, the superclass clause can contain nested sends, and these would come from the nested class accessor as well. And the accessor has its own stuff, which needs to be filtered out. It will be tad ugly to get this right.
I'd tend to add comments to that effect here, rather than waiting to fix it before it gets in.